src/database/Database.h

```diff
@@ -214,6 +218,16 @@ class Database : NonMovableOrCopyable
     // Access the optional SOCI connection pool available for worker
     // threads. Throws an error if !canUsePool().
     soci::connection_pool& getPool();
+
+  protected:
+    // Give clients the opportunity to perform operations on databases while
```

Comment:
the application test hook that you removed here should just be removed from history as you never used it (it's just extra commits and churn for no reason)

Reply:
Done.